src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequest.java

```diff
@@ -19,24 +19,38 @@
 import org.opensearch.common.bytes.BytesReference;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.rest.RestRequest;
 
 public class UploadGeoJSONRequest extends ActionRequest {
 
+    private final RestRequest.Method method;
     private BytesReference content;
 
-    public UploadGeoJSONRequest(BytesReference content) {
-        this.content = Objects.requireNonNull(content);
+    public UploadGeoJSONRequest(RestRequest.Method method, BytesReference content) {
+        super();
+        this.method = Objects.requireNonNull(method, "method cannot be null");
+        this.content = Objects.requireNonNull(content, "content cannot be null");
     }
 
     public UploadGeoJSONRequest(StreamInput in) throws IOException {
         super(in);
         this.content = Objects.requireNonNull(in.readBytesReference());
+        this.method = in.readEnum(RestRequest.Method.class);
```

Comment:
For consistency, should this be wrapped with requireNonNull?

Reply:
Ack